### ISSUE TITLE
Do not pass tests on 32-bit architectures (armh, i586) in qcheck 0.18

### ISSUE THREAD

**Issue by RiderALT** [c-cube/qcheck#152]
Environment:
ALT Linux Sisyphus, ocaml-4.12, ounit-2.2.4. qcheck-0.18

```
[builder@localhost ocaml-qcheck-0.18]$ dune runtest
(cd _build/default && /usr/bin/diff -u example/ounit/output.txt.expected example/ounit/output.txt)
--- example/ounit/output.txt.expected   2021-08-13 09:05:24.730647851 +0000
+++ example/ounit/output.txt    2021-08-13 09:05:24.729647856 +0000
@@ -6,7 +6,7 @@
 
 
 test `error_raise_exn` raised exception `Dune__exe__QCheck_ounit_test.Error`
-on `0 (after 63 shrink steps)`
+on `0 (after 31 shrink steps)`
 
 ------------------------------------------------------------------------------
 ==============================================================================
        test alias test/core/runtest
Testing `QCheck'.
This run has ID `4368B7F1-ADEC-40ED-B472-EB582386B77D'.

  [OK]          Shrink          0   int_towards.
  [OK]          Shrink          1   int32_towards.
  [OK]          Shrink          2   int64_towards.
  [OK]          Shrink          3   float_towards.
  [OK]          Gen             0   opt with default ratio.
  [OK]          Gen             1   opt with custom ratio.
  [OK]          Test            0   make with custom count.
  [OK]          Test            1   make with custom count.
  [OK]          Test            2   make with default count.
  [OK]          Test            3   make with env count.
  [OK]          String          0   shrinking.

Full test results in `~/RPM/BUILD/ocaml-qcheck-0.18/_build/default/test/core/_build/_tests/QCheck'.
Test Successful in 0.142s. 11 tests run.
Done: 88/93 (jobs: 1)File "example/output.txt.expected", line 1, characters 0-0:
        diff (internal) (exit 1)
(cd _build/default && /usr/bin/diff -u example/output.txt.expected example/output.txt)
--- example/output.txt.expected 2021-08-13 09:05:29.643625563 +0000
+++ example/output.txt  2021-08-13 09:05:29.637625590 +0000
@@ -8,7 +8,7 @@
 
 === Error ======================================================================
 
-Test should_error_raise_exn errored on (63 shrink steps):
+Test should_error_raise_exn errored on (31 shrink steps):
 
 0
 
@@ -59,9 +59,9 @@
 
 --- Failure --------------------------------------------------------------------
 
-Test FAIL_pred_map_commute failed (127 shrink steps):
+Test FAIL_pred_map_commute failed (67 shrink steps):
 
-([3], {_ -> 0}, {3 -> false; _ -> true})
+([0], {_ -> -5}, {-5 -> true; _ -> false})
 
 --- Failure --------------------------------------------------------------------
 
@@ -90,13 +90,13 @@
 
 --- Failure --------------------------------------------------------------------
 
-Test long_shrink failed (149 shrink steps):
+Test long_shrink failed (83 shrink steps):
 
 ([0], [-1])
 
 --- Failure --------------------------------------------------------------------
 
-Test mod3_should_fail failed (84 shrink steps):
+Test mod3_should_fail failed (34 shrink steps):
 
 -21
 
@@ -129,7 +129,7 @@
 
 Warning for test WARN_unlikely_precond:
 
-WARNING: only 0.5% tests (of 2000) passed precondition for "WARN_unlikely_precond"
+WARNING: only 0.6% tests (of 2000) passed precondition for "WARN_unlikely_precond"
 
 NOTE: it is likely that the precondition is too strong, or that the generator is buggy.
 
@@ -137,7 +137,7 @@
 
 Test FAIL_unlikely_precond failed:
 
-ERROR: only 0.5% tests (of 2000) passed precondition for "FAIL_unlikely_precond"
+ERROR: only 0.6% tests (of 2000) passed precondition for "FAIL_unlikely_precond"
 
 NOTE: it is likely that the precondition is too strong, or that the generator is buggy.
 
@@ -292,27 +292,27 @@
 +++ Stats for stat_display_test_7 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 
 stats dist:
-  num: 100000, avg: 2541076923587387.50, stddev: 2660730801206827008.00, median 158655268318060, min -4611522359435274428, max 4611540922436307689
-  -4611522359435274428..-4150369195341695293: #####################################################          4976
-  -4150369195341695292..-3689216031248116157: #####################################################          4963
-  -3689216031248116156..-3228062867154537021: ######################################################         5038
-  -3228062867154537020..-2766909703060957885: #####################################################          4979
-  -2766909703060957884..-2305756538967378749: #####################################################          5001
-  -2305756538967378748..-1844603374873799613: #####################################################          4982
-  -1844603374873799612..-1383450210780220477: #####################################################          5025
-  -1383450210780220476.. -922297046686641341: ####################################################           4901
-   -922297046686641340.. -461143882593062205: #######################################################        5126
-   -461143882593062204..       9281500516931: #####################################################          5008
-         9281500516932..  461162445594096067: ######################################################         5041
-    461162445594096068..  922315609687675203: #####################################################          5001
-    922315609687675204.. 1383468773781254339: #####################################################          4986
-   1383468773781254340.. 1844621937874833475: #####################################################          4949
-   1844621937874833476.. 2305775101968412611: #####################################################          5025
-   2305775101968412612.. 2766928266061991747: #####################################################          5022
-   2766928266061991748.. 3228081430155570883: #####################################################          4958
-   3228081430155570884.. 3689234594249150019: #####################################################          4998
-   3689234594249150020.. 4150387758342729155: #####################################################          4982
-   4150387758342729156.. 4611540922436308291: ######################################################         5039
+  num: 100000, avg: 336840.90, stddev: 619301756.02, median 895228, min -1073728193, max 1073739280
+  -1073728193.. -966354820: #####################################################          5009
+   -966354819.. -858981446: ####################################################           5004
+   -858981445.. -751608072: ####################################################           4917
+   -751608071.. -644234698: #####################################################          5028
+   -644234697.. -536861324: ####################################################           4962
+   -536861323.. -429487950: #####################################################          5039
+   -429487949.. -322114576: ####################################################           4927
+   -322114575.. -214741202: #####################################################          5054
+   -214741201.. -107367828: #####################################################          5065
+   -107367827..       5546: ####################################################           4954
+         5547..  107378920: ####################################################           4943
+    107378921..  214752294: ###################################################            4900
+    214752295..  322125668: ######################################################         5126
+    322125669..  429499042: #######################################################        5198
+    429499043..  536872416: ####################################################           4988
+    536872417..  644245790: ####################################################           4940
+    644245791..  751619164: ####################################################           5002
+    751619165..  858992538: ####################################################           4928
+    858992539..  966365912: #####################################################          5070
+    966365913.. 1073739286: ####################################################           4946
 ================================================================================
 1 warning(s)
 failure (9 tests failed, 1 tests errored, ran 24 tests)
```

**Comment by c-cube** [comment]
to be fair, none of these seem to be actual bugs, just a different behavior because integers are different. I'm not sure whether we should try to fix this (by having a distinct `output.txt.expected` files) or to just disable tests on 32 bit architectures?

**Comment by RiderALT** [comment]
Yes indeed. It also seemed to me that these errors are not real bugs. But this is a regression in relation to version 0.17, so I found it necessary to report this ISSUE.

It would be great if I did not have to turn off tests on a 32-bit architecture. Perhaps you need two sets for different architectures?

**Comment by c-cube** [comment]
it's a regression only in the sense that we have actual tests now :-).

**Comment by c-cube** [comment]
if you know how to build CI on 32 bits, I'd accept a PR to test them. Otherwise I'll look into a way of disabling the tests on 32 bits.

**Comment by jmid** [comment]
Fixed by #250
